modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/AmqpChannel.java

```diff
@@ -364,6 +375,45 @@ public AmqpDeliverMessage createDeliverMessage(Message message, ShortString cons
         return new AmqpDeliverMessage(message, consumerTag, this, queueName, broker);
     }
 
+    /**
+     * Start local transaction on the channel
+     */
+    public void setLocalTransactional() {
+        transaction = new LocalTransaction(broker);
+        messageAggregator.setTransaction(transaction);
+    }
+
+    /**
+     * Commit the transaction on the channel
+     */
+    public void commit() throws ChannelException {
+        if (!isTransactional()) {
+            throw new ChannelException(ChannelException.PRECONDITION_FAILED,
```

Comment:
We can move this exception to auto commit transaction commit IMO

Reply:
Please find the changes in d1ebb42